src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -544,12 +544,19 @@ where
             let pkg = Box::new(source).download_now(pkgid, config)?;
             Ok(pkg)
         }
-        None => bail!(
-            "could not find `{}` in {} with version `{}`",
-            dep.package_name(),
-            source.source_id(),
-            dep.version_req(),
-        ),
+        None => {
+            for pkg_id in deps.iter() {
+                if source.is_yanked(pkg_id.package_id()).unwrap_or(false) {
+                    bail!("provided package has been yanked `{}`", pkg_id.package_id())
```

Comment:
Just a small note on the wording, perhaps it could be written as:

```cannot install package `{}`, it has been yanked from {}```

Reply:
Thanks for suggestion, I'll definitely update the message and yes can please help me out to add test for this?